cnxpublishing/views/admin.py

```diff
@@ -271,3 +295,266 @@ def admin_edit_site_message_POST(request):
+                INNER JOIN modules AS m USING (module_ident)
+                LEFT JOIN print_style_recipes as ps
+                    ON ps.print_style=m.print_style
+                LEFT JOIN latest_modules as lm
+                ON lm.uuid=m.uuid
+                WHERE m.uuid=%s ORDER BY bpsa.created DESC;
+                """, vars=(uuid,))
+            modules = cursor.fetchall()
+            if len(modules) == 0:
+                raise httpexceptions.HTTPBadRequest(
+                    '{} is not a book'.format(uuid))
+
+            states = []
+            collection_info = modules[0]
+
+            for row in modules:
+                message = ''
+                result_id = row[-1]
+                result = AsyncResult(id=result_id)
+                if result.failed():  # pragma: no cover
+                    message = result.traceback
+                latest_recipe = row[4]
+                current_recipe = row[5]
+                latest_version = row[6]
+                current_version = row[7]
+                state = result.state
+                if current_recipe != latest_recipe:
+                    state += ' stale_recipe'
+                if current_version != latest_version:
+                    state += ' stale_content'
+                states.append({
+                    'version': row[7],
+                    'recipe': row[5],
+                    'created': str(row[-2]),
+                    'state': state,
+                    'state_message': message,
+                })
+
+    return {'uuid': str(collection_info[2]),
+            'title': collection_info[0].decode('utf-8'),
+            'authors': format_autors(collection_info[1]),
+            'print_style': collection_info[3],
+            'current_recipe': collection_info[4],
+            'current_ident': collection_info[8],
+            'current_state': states[0]['state'],
+            'states': states}
+
+
+@view_config(route_name='admin-content-status-single', request_method='POST',
+             renderer='templates/content-status-single.html',
+             permission='administer')
+def admin_content_status_single_POST(request):
+    """ Retriggers baking for a given book. """
+    args = admin_content_status_single(request)
+    title = args['title']
+    if args['current_state'] == 'SUCCESS':
+        args['response'] = title + ' is not stale, no need to bake'
+        return args
+
+    settings = request.registry.settings
+    with psycopg2.connect(settings[config.CONNECTION_STRING]) as db_conn:
+        with db_conn.cursor() as cursor:
+            cursor.execute("SELECT stateid FROM modules WHERE module_ident=%s",
+                           vars=(args['current_ident'],))
+            data = cursor.fetchall()
+            if len(data) == 0:
+                raise httpexceptions.HTTPBadRequest(
+                    'invalid module_ident: {}'.format(args['current_ident']))
+            if data[0][0] == 5:
```

Comment:
I think if the state is `5` (post-publication) or `6` (processing), we shouldn't reset the state.